parser/parser/cc/driver.cc

```diff
@@ -63,7 +63,7 @@ void base_driver::rewind_if_dedented(token_t token, token_t kEND, bool force) {
 
 typedruby_release::typedruby_release(std::string_view source, sorbet::StableStringStorage<> &scratch,
                                      const struct builder &builder, bool traceLexer, bool indentationAware)
-    : base_driver(ruby_version::RUBY_27, source, scratch, builder, traceLexer, indentationAware) {}
+    : base_driver(ruby_version::RUBY_31, source, scratch, builder, traceLexer, indentationAware) {}
```

Comment:
Does changing the default version to `RUBY_31` change behavior in a way that would cause problems for a codebase that's still on 2.7?

Reply:
Not currently since the changes made in lexer are backwards compatible  and parser tests pass with `RUBY_27`.

There's still a risk of issues being introduced in the future though since CI will not test with `RUBY_27`. I don't think this risk can be eliminated without supplying the actual ruby version number to the lexer instead of a fixed version.